modules/core/src/main/java/org/apache/synapse/unittest/Assertor.java

```diff
@@ -0,0 +1,459 @@
+                        isAssert = expected.equals(mediatedResult);
+
+                        break;
+
+                    default:
+                        isAssert = false;
+                        mediatedResult = message;
+                        message = "Received assert actual value for sequences not defined";
+                }
+
+                logger.info("Sequence Assert Actual - " + actual);
+                logger.info("Sequence Assert Expected - " + expected);
+                logger.info("Sequence mediated result for actual - " + mediatedResult);
+                if (isAssert) {
+                    logger.info("Sequence assertEqual for " + actualType[0] + " type passed successfully");
+                } else {
+                    isAssertEqualFailed = true;
+                    messageOfAssertEqual = message;
+                    logger.error("Sequence assertEqual for " + actualType[0] + " type failed - " + message + "\n");
+                }
+            }
+
+        }
+
+        logger.info("AssertEquals assertion success - " + !isAssertEqualFailed);
+        return new Pair<>(!isAssertEqualFailed, messageOfAssertEqual);
+    }
+
+    /**
+     * Method of assertionNotNull for Sequence test cases.
+     *
+     * @param assertNotNull array of assertNotNull
+     * @param msgCtxt       message context
+     * @return Pair<Boolean, String> which has status of assertNotNull and message if any error occurred
+     */
+    private static Pair<Boolean, String> startAssertNotNullsForSequence(List<AssertNotNull> assertNotNull,
+                                                                        MessageContext msgCtxt) {
+        logger.info("\n");
+        logger.info("---------------------Assert Not Null---------------------\n");
+        boolean isAssertNotNullFailed = false;
+        String messageOfAssertNotNull = null;
+
+        for (AssertNotNull assertItem : assertNotNull) {
+
+            if (!isAssertNotNullFailed) {
+
+                String actual = assertItem.getActual();
+                String message = assertItem.getMessage();
+                boolean isAssertNull;
+                String[] actualType = actual.split(":");
+                String mediatedResult;
+
+                switch (actualType[0]) {
+
+                    case INPUT_PROPERTY_BODY:
+                        mediatedResult = Trimmer.trimStrings(
+                                msgCtxt.getEnvelope().getBody().getFirstElement().toString());
+                        isAssertNull = mediatedResult.isEmpty();
+
+                        break;
+
+                    case INPUT_PROPERTY_CONTEXT:
+                        mediatedResult = Trimmer.trimStrings(msgCtxt.getProperty(actualType[1]).toString());
+                        isAssertNull = mediatedResult.isEmpty();
+
+                        break;
+
+                    case INPUT_PROPERTY_AXIS2:
+                        Axis2MessageContext axis2smc = (Axis2MessageContext) msgCtxt;
```

Comment:
Please use a meaningful name for `axis2smc`

Reply:
Done